pkg/util/jsonpatch/jsonpatch.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jsonpatch
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func FieldExistsByJSONPath(object interface{}, path string) bool {
+	pathParts := strings.Split(path, "/")
+	// invalid path
+	if len(pathParts) < 2 || len(pathParts) > 0 && pathParts[0] != "" {
+		return false
+	}
+	return fieldExistsByJSONPathHelper(object, pathParts[1:])
+}
+
+func fieldExistsByJSONPathHelper(object interface{}, pathParts []string) bool {
+	if object == nil {
+		return false
+	}
+
+	t := reflect.TypeOf(object)
+	v := reflect.ValueOf(object)
+
+	if v.Kind() == reflect.Pointer {
+		t = t.Elem()
+		v = v.Elem()
+	}
+
+	if t.Kind() != reflect.Struct {
+		return false
+	}
+
+	for i := 0; i < t.NumField(); i++ {
+		ft := t.Field(i)
+		fv := v.Field(i)
+
+		tag := ft.Tag.Get("json")
+		if tag == "" {
+			tag = ft.Name
+		} else if tagParts := strings.Split(tag, ","); len(tagParts) > 1 {
```

Comment:
Yeah, most of all tests using omitempty.

Reply:
but I guess in reality fields without omitempty are legal, so we need to make sure this works well, take a look at some declarations here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/types.go as a source of inspiration for additional test cases. For example: https://github.com/kubernetes/kubernetes/blob/b8e57a1c201914e8c5164a165f446480c7ff0bba/staging/src/k8s.io/api/core/v1/types.go#L965

Reply:
> why len(tagParts) > 1

User can use `json:"bar,omitempty"`. But we just need `bar` value.

Reply:
But if we have `json:"bar` we  need to do nothing because we already get tag on `tag := ft.Tag.Get("json")`

Reply:
ah, got it, I got tricked by assigning `tag := ft.Tag.Get("json")` while this is not necessarily the tag. Maybe it is cleaner to read if  `jsonTag := ft.Tag.Get("json")`, then instead calling `tag` call name it `fieldName`, or `fName` for short. It could be inside a helper function.